app/controllers/supervisor/comments_controller.rb

```diff
@@ -0,0 +1,19 @@
+class Supervisor::CommentsController < Supervisor::BaseController
+
+  def create
+    comment = Comment.new(comment_params)
+
+    if comment.save
```

Comment:
I would just write 

    CommentMailer.email(comment).deliver_later if comment.text.present? 

Imho the controller is the ideal place to start actions like this. The model is not. Callbacks in the model should only be used to keep the data-model consistent or simpler, like setting a flag on another model, or setting default values, or calculating values. Indeed adding a callback (or changing the `set_checked`) to send the e-mail breaks the single responsibility principle. 

However, if changing or setting an attribute needs to fire an external action (e.g. an API call or mail) and this attribute change can come from different places (multiple controller actions), then I use an observer. 

The objective imho is to make the system as clear as possible, and keep the action as close as possible to the change. A controller changes something: keep the action (send the mail) there. It is easier to reason about what action is causing what.